### PR TITLE
Add EX version 3 with time sequences, templates

### LINKED ISSUE #208: Support I/O of time sequence parameters in EX format
EX I/O is currently limited to writing parameters at a single time.

Propose to add a time sequence construct which can be selected in node field header to read/write a time sequence of parameters.

This will require an increment of EX version to 3.

### LINKED ISSUE #212: Support node and element template reuse in EX format
EX files are pathologically large when alternate nodes and elements have a new header.
This item will allow node and element templates to be defined once and reused by name.
Plan to add this with EX version 3.

### PR DESCRIPTION
**Pull request by rchristie**
Closes #208.
Closes #212.

### CODE REVIEW

**Review by alan-wu — COMMENTED** [review]
I like the new format.

Like the new readFileToString function for testing.

Great to see the FEtime sequence to be class like.

It looks good but I skinned through most of it.

File: tests/graphics/sceneviewer.cpp [hunk 1]
```diff
@@ -501,47 +503,52 @@ TEST(ZincSceneviewer, description_io)
 {
 	ZincTestSetupCpp zinc;
 
+	std::cerr << "1.\n";
```
Comment (hsorby) [inline tests/graphics/sceneviewer.cpp]: There are a few of these what I presume to be debugging messages that need to be removed.

File: tests/graphics/scene.cpp [hunk 2]
```diff
@@ -690,6 +691,8 @@ TEST(cmzn_scene, threejs_export_texture_cpp)
 
 	int result;
 
+	std::cerr << "1.\n";
```
Comment (hsorby) [inline tests/graphics/scene.cpp]: Debugging messages need to be removed.

**Review by hsorby — COMMENTED** [review]
I too have had a quick scan through and have tested this locally and haven't found any issues.  Other than cleaning up the debug messages to std::err I think this will be good to merge.

File: src/finite_element/export_finite_element.h [hunk 3]
```diff
@@ -63,33 +63,41 @@ Global/Public functions
...
  * @param region  The region to output nodes/elements/data from.
- * @param group  Optional subgroup to output.
- * @param root_region  The root region output paths are relative to.
- * @param file_name  Name of file. 
+ * @param groupName  Optional name of group to limit output to.
+ * @param writeDomainTypes  Bitwise OR of cmzn_field_domain_type flags controlling
+ * which meshes or nodesets to write.
+ * @param timeSet  True if following time is set to be output, otherwise false.
+ * @param time  Time to output parameters at if timeSet is true. Nearest time in
+ * time sequences is output if not an exact match.
```
Comment (rchristie) [inline src/finite_element/export_finite_element.h]: Check - I think this is interpolated now.

File: tests/graphics/sceneviewer.cpp (hunk 1 above)
Comment (rchristie) [reply tests/graphics/sceneviewer.cpp]: Done.

File: src/finite_element/export_finite_element.h (hunk 3 above)
Comment (rchristie) [reply src/finite_element/export_finite_element.h]: Fixed comment.

**Review by hsorby — APPROVED** [review]

**Pull request merged**